backend/src/main/java/com/yohoard/notes/TagService.java

```diff
@@ -0,0 +1,39 @@
+package com.yohoard.notes;
+
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class TagService {
+
+    private final TagRepository tagRepository;
+
+    public TagService(TagRepository tagRepository) {
+        this.tagRepository = tagRepository;
+    }
+
+    public List<Tag> getOrCreate(List<String> names) {
```

Comment:
Myślę, że getOrCreate najlepiej by było rozdzielić, to nie jest case gdzie te operacje nie mają aż takiego znaczenia. getOrCreate dla Listy sugeruje że na raz możesz stworzyć pierdyliard tagów i je tak po prostu z kolana wepchnąć do bazy. WIęc albo jakaś walidacja po drodze, albo user będzie mógł radośnie bałaganić :D